src/main/java/org/jabref/logic/git/GitHandler.java

```diff
@@ -171,34 +171,80 @@ public void mergeBranches(String targetBranch, String sourceBranch, MergeStrateg
+        try {
+            Git git = Git.open(this.repositoryPathAsFile);
+            String remoteURL = git.getRepository().getConfig().getString("remote", "origin", "url");
+            Boolean isSshRemoteRepository = remoteURL != null ? remoteURL.contains(".git") : false;
+            
+            git.verifySignature();
+
+            if (isSshRemoteRepository) {
+                TransportConfigCallback transportConfigCallback = new SshTransportConfigCallback();
                 git.push()
-                   .setCredentialsProvider(credentialsProvider)
-                   .call();
-            } catch (GitAPIException e) {
+               .setTransportConfigCallback(transportConfigCallback)
+               .call();
+            } else {
+                String gitUsername = "";
+                String gitPassword = "";
+
+                if (dialogService != null) {
+                    gitUsername = dialogService.showInputDialogAndWait(Localization.lang("Git credentials"), Localization.lang("git username")).get();
+                    gitPassword = dialogService.showPasswordDialogAndWait(Localization.lang("Git credentials"), Localization.lang("password"), Localization.lang("password")).get();
+
+                    UsernamePasswordCredentialsProvider credentialsProvider = new UsernamePasswordCredentialsProvider(gitUsername, gitPassword);
+
+                    git.push()
+                    .setCredentialsProvider(credentialsProvider)
+                    .call();
+                } else {
+                    git.push()
+                    .call();
+                }
+            }
+            
+        } catch (IOException | GitAPIException e) {
+            if (e.getMessage().equals("origin: not found.")) {
+                LOGGER.info("No remote repository detected. Push skiped.");
+            } else {
                 LOGGER.info("Failed to push");
+                throw new RuntimeException(e);
             }
         }
     }
 
+    /**
+     * Pulls all commits made to the branch that is tracked by the currently checked out branch.
+     * If pulling to remote fails, it fails silently.
+     */
     public void pullOnCurrentBranch() throws IOException {
         try (Git git = Git.open(this.repositoryPathAsFile)) {
             try {
                 git.pull()
-                   .setCredentialsProvider(credentialsProvider)
                    .call();
             } catch (GitAPIException e) {
-                LOGGER.info("Failed to push");
+                if (e.getMessage().equals("origin: not found")) {
+                    LOGGER.info("No remote repository detected. Push skiped.");
+                } else {
+                    LOGGER.info("Failed to pull");
+                    throw new RuntimeException(e);
+                }
             }
         }
     }
 
+    /**
+     * Get currently checked out branch.
+     * If checking out fails, it fails silently.
```

Comment:
I don't see any checkout call there. -- Maybe you mean something else? Maybe demonstrate using a `@Test`?